### PR TITLE
#12366 - Add facility to external messages

### LINKED ISSUE #12366: Extend external message with person address facility reference [2]
### Feature Description

For eSanté, it's requested that an existing facility in the system with the same address as the tested person is automatically added to the person's address as it can be assumed that the person is living in that facility.

In order to do this, the external message needs to be extended with a facility reference field, a facade method that searches for facilities with matching addresses is required, and the external message processing logic has to be extended in order to actually add the facility from the external message to the person's address.

### Added Value/Benefit

Users won't have to manually link the facility to the person because we can assume that the person is living in the facility.

### Acceptance Criteria

- [x] ExternalMessageDto has been extended with a person address facility reference field
- [x] FacilityFacade has been extended with a `getFacilityByAddress` field that considers street, postal code, and city
- [x] The processing logic for external messages has been extended to take over the facility reference from the external message to the person's address

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by MateStrysewske**
Fixes #12366

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12482

File: sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/ExternalMessage.java [hunk 1]
```diff
@@ -337,6 +340,15 @@ public void setPersonCountry(Country personCountry) {
 		this.personCountry = personCountry;
 	}
 
+	@OneToOne(fetch = FetchType.LAZY)
+	public Facility getPersonFacility() {
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/ExternalMessage.java:344]: Should be `@ManyToOne`

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12482

**Review by leventegal-she — APPROVED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12482

**Pull request merged**